toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/Authenticator.kt

```diff
@@ -38,21 +37,98 @@ import com.arcgismaps.httpcore.authentication.OAuthUserConfiguration
 import com.arcgismaps.httpcore.authentication.OAuthUserSignIn
 
 /**
- * Displays appropriate Authentication UI when issued a challenge. For example, if an [ArcGISAuthenticationChallenge]
- * is issued and the [AuthenticatorState] has a corresponding [OAuthUserConfiguration],
- * then a Custom Chrome Tab will be launched to complete the OAuth sign in.
+ * Displays appropriate Authentication UI when a challenge is issued.
  *
- * @param authenticatorState an [AuthenticatorState]. See [AuthenticatorState.Companion.Factory].
+ * For example, when an [ArcGISAuthenticationChallenge] is issued and the [AuthenticatorState] has a corresponding
+ * [OAuthUserConfiguration], then a Custom Tab will be launched to complete the OAuth sign in.
+ *
+ * All Authentication components will be displayed in full screen. See [DialogAuthenticator] for alternate behavior.
+ *
+ * @param authenticatorState the application viewModel's [AuthenticatorState].
+ * @param modifier the [Modifier] to apply to this Authenticator.
  * @param onPendingOAuthUserSignIn if not null, this will be called when an OAuth challenge is pending
  * and the browser should be launched. Use this if you wish to handle OAuth challenges from your own
  * activity rather than using the [OAuthUserSignInActivity].
+ * @see DialogAuthenticator
  * @since 200.2.0
  */
 @Composable
 public fun Authenticator(
     authenticatorState: AuthenticatorState,
-    modifier: Modifier = Modifier.fillMaxSize(),
+    modifier: Modifier = Modifier,
     onPendingOAuthUserSignIn: ((OAuthUserSignIn) -> Unit)? = null
+) {
+    AuthenticatorDelegate(
+        authenticatorState = authenticatorState,
+        modifier = modifier.fillMaxSize(),
+        onPendingOAuthUserSignIn = onPendingOAuthUserSignIn
+    )
+}
+
+/**
+ * Displays appropriate Authentication UI when a challenge is issued.
+ *
+ * For example, when an [ArcGISAuthenticationChallenge] is issued and the [AuthenticatorState] has a corresponding
+ * [OAuthUserConfiguration], then a Custom Tab will be launched to complete the OAuth sign in.
+ *
+ * The [ServerTrustAuthenticator] and [UsernamePasswordAuthenticator] components will be displayed in an [AlertDialog].
```

Comment:
`ServerTrustAuthenticator` & `UsernamePasswordAuthenticator` are internal types, so we cannot link to them in public doc. Instead I suggest:
```suggestion
 * Server trust prompts and username/password prompts will be displayed in an [AlertDialog].
```